cprover_bindings/src/goto_program/expr.rs

```diff
@@ -1448,6 +1456,7 @@ impl Expr {
 
     /// `ArithmeticOverflowResult r; >>>r.overflowed = builtin_sub_overflow(self, e, &r.result)<<<`
     pub fn mul_overflow(self, e: Expr) -> ArithmeticOverflowResult {
+        // TODO: Should we always use the one below?
         let result = self.clone().mul(e.clone());
         let overflowed = self.mul_overflow_p(e);
         ArithmeticOverflowResult { result, overflowed }
```

Comment:
I'd answer that "TODO" with a "yes, unless there really is a need to only compute the Boolean would-this-operation-overflow result." That's not the case here, which does the multiplication, so that's likely just a left-over from the time where we didn't yet have the OverflowResult* expressions. We should get rid of this.

Reply:
Yes, the intention was to replace all the usage of `mul_overflow` with `mul_overflow_result` and get rid of `mul_overflow`.